system/loggerd/tests/test_loggerd.py

```diff
@@ -276,6 +263,16 @@ def test_rlog(self):
       sent.clear_write_flag()
       self.assertEqual(sent.to_bytes(), m.as_builder().to_bytes())
 
+  def test_preserving_flagged_segments(self):
+    services = set(random.sample(CEREAL_SERVICES, random.randint(5, 10))) | {"userFlag"}
+
+    managed_processes["loggerd"].start()
+    self._publish_random_messages(services)
+    managed_processes["loggerd"].stop()
```

Comment:
the loggerd start/stop and services can even be moved into _publish_random_messages